tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -1187,18 +1199,8 @@ private void validateTitleChanged(String expectedTitle, Runnable browserSetFunc)
 	browserSetFunc.run();
 	shell.open();
 
-	boolean passed = waitForPassCondition(() -> actualTitle.get().equals(expectedTitle));
-	String errMsg = "";
-	if (!passed) {
-		if (actualTitle.get().length() == 0) {
-			errMsg = "Test timed out. TitleListener not fired";
-		} else {
-			errMsg = "\nExpected title and actual title do not match."
-					+ "\nExpected: " + expectedTitle
-					+ "\nActual: " + actualTitle;
-		}
-	}
-	assertTrue(errMsg + testLog.toString(), passed);
+	assertTrue("Test timed out. TitleListener not fired", waitForPassCondition(() -> actualTitle.get().equals(expectedTitle)));
+	assertEquals(testLog.toString(), expectedTitle, actualTitle.get());
```

Comment:
The way it is written right now you're never going to see the 2nd assertion (`assertEquals`) fail because you are checking the same thing in the 1st assertion (`assertTrue`).

You're also not contemplating the (old) case: the title is empty.

```suggestion
	waitForPassCondition(() -> actualTitle.get().equals(expectedTitle));
	assertTrue("Test timed out. TitleListener not fired", actualTitle.get().length() != 0);
	assertEquals(testLog.toString(), expectedTitle, actualTitle.get());
```

Reply:
Done